lib/archethic/db.ex

```diff
@@ -62,12 +62,7 @@ defmodule Archethic.DB do
 
   @callback transaction_exists?(binary()) :: boolean()
 
-  @callback register_p2p_summary(
-              node_public_key :: Crypto.key(),
-              date :: DateTime.t(),
-              available? :: boolean(),
-              average_availability :: float()
-            ) :: :ok
+  @callback register_p2p_summary(list()) :: :ok
```

Comment:
You should include which kind of information is present in the list.
This way the implementations will know which data must be handled.